presto-hive/src/main/java/com/facebook/presto/hive/metastore/HivePrivilegeInfo.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive.metastore;
+
+import com.facebook.presto.spi.security.Privilege;
+import com.google.common.collect.ImmutableSet;
+import org.apache.hadoop.hive.metastore.api.PrivilegeGrantInfo;
+
+import java.util.Arrays;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static com.facebook.presto.hive.metastore.HivePrivilegeInfo.HivePrivilege.DELETE;
+import static com.facebook.presto.hive.metastore.HivePrivilegeInfo.HivePrivilege.INSERT;
+import static com.facebook.presto.hive.metastore.HivePrivilegeInfo.HivePrivilege.OWNERSHIP;
+import static com.facebook.presto.hive.metastore.HivePrivilegeInfo.HivePrivilege.SELECT;
+import static com.facebook.presto.hive.metastore.HivePrivilegeInfo.HivePrivilege.UPDATE;
+import static java.util.Locale.ENGLISH;
+
+public class HivePrivilegeInfo
+{
+    HivePrivilegeInfo(HivePrivilege hivePrivilege, boolean grantOption)
+    {
+        this.hivePrivilege = hivePrivilege;
+        this.grantOption = grantOption;
+    }
+
+    public HivePrivilege getHivePrivilege()
+    {
+        return hivePrivilege;
+    }
+
+    public void setHivePrivilege(HivePrivilege hivePrivilege)
+    {
+        this.hivePrivilege = hivePrivilege;
+    }
+
+    public boolean isGrantOption()
+    {
+        return grantOption;
+    }
+
+    public void setGrantOption(boolean grantOption)
+    {
+        this.grantOption = grantOption;
+    }
+
+    public enum HivePrivilege
+    {
+        SELECT, INSERT, UPDATE, DELETE, OWNERSHIP;
+    }
+
+    private HivePrivilegeInfo.HivePrivilege hivePrivilege;
```

Comment:
Now that I think about it, even if you want your class to be mutable you can still make all the fields final and when you want to change a given field you end up re-building the entire object. I don't think this piece of code will be in a performance critical code path so such object copying shouldn't be a problem. There are idioms in Java for creating copy constructors, you'll have to check Effective Java because I can't remember it off the top of my head.

Leave the class as it is for now and we'll see what Dain says.

In general, always aim to have as much immutable state as possible.